test/e2e/specs/editor/various/multi-block-selection.spec.js

```diff
@@ -0,0 +1,1372 @@
+		await page.keyboard.press( 'ArrowDown' );
+
+		const [ paragraph1, paragraph2 ] = await editor.canvas
+			.getByRole( 'document', { name: 'Paragraph block' } )
+			.all();
+
+		await paragraph1.hover();
+		await page.mouse.down();
+		await paragraph2.hover();
+		await page.mouse.up();
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedFlatIndices )
+			.toEqual( [ 1, 2 ] );
+	} );
+
+	test( 'should cut and paste', async ( { editor, pageUtils } ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '1' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '2' },
+		} );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'x' );
+
+		await expect.poll( editor.getBlocks ).toEqual( [] );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'v' );
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{ name: 'core/paragraph', attributes: { content: '1' } },
+			{ name: 'core/paragraph', attributes: { content: '2' } },
+		] );
+	} );
+
+	test( 'should copy and paste', async ( { page, editor, pageUtils } ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'first paragraph' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'second paragraph' },
+		} );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'c' );
+
+		await page.keyboard.press( 'Backspace' );
+		await expect
+			.poll( editor.getBlocks, 'should select all blocks after copying' )
+			.toEqual( [] );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'v' );
+		await expect
+			.poll( editor.getBlocks, 'should copy and paste multiple blocks' )
+			.toMatchObject( [
+				{ attributes: { content: 'first paragraph' } },
+				{ attributes: { content: 'second paragraph' } },
+			] );
+
+		await page.keyboard.type( '|' );
+		await expect
+			.poll(
+				editor.getBlocks,
+				'should place the caret at the end of last pasted paragraph'
+			)
+			.toMatchObject( [
+				{ attributes: { content: 'first paragraph' } },
+				{ attributes: { content: 'second paragraph|' } },
+			] );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'v' );
+		await page.keyboard.type( '|' );
+		await expect
+			.poll( editor.getBlocks, 'should replace blocks' )
+			.toMatchObject( [
+				{ attributes: { content: 'first paragraph' } },
+				{ attributes: { content: 'second paragraph|' } },
+			] );
+		await page.keyboard.press( 'Backspace' );
+
+		for ( let i = 0; i < 3; i += 1 ) {
+			await page.keyboard.press( 'ArrowLeft' );
+		}
```

Comment:
Should we use the `pressKeyTimes` here?

```suggestion
		await pageUtils.pressKeyTimes( 'ArrowLeft', 3 );
```

Reply:
Oh, yeah! Nice catch.